krabs/krabs/guid.hpp

```diff
@@ -309,4 +309,21 @@ namespace std
 
         return { managed.get() };
     }
+
+    template<>
+    struct std::hash<krabs::guid>
+    {
+        size_t operator()(const krabs::guid& guid) const
```

Comment:
I know this PR is already merged but FWIW...

I find it strange to calculate a byte-by-byte hash on something that is already highly random. A good specialisation of std::hash just has to have low risk of collision. So if you just take the lower sizeof(size_t) bits of the 128-bit GUID you’ve already got a very unique value, i.e. just return:

reinterpret_cast<const size_t*>(&guid)[(sizeof(guid) / sizeof(size_t)) - 1]

Reply:
That's a good point.
I was taking a look at the .NET guid hash which at least treats it as blocks and uses the whole value. May be worthwhile.
There are different guid versions with varying amounts of randomness in the different parts - but given that these are all provider guids, I'd guess they are all the random kind.
```
        public override int GetHashCode()
        {
            return _a ^ (((int)_b << 16) | (int)(ushort)_c) ^ (((int)_f << 24) | _k);
        }```

Reply:
Good catch, I applied the recommended improvement in #144